File: misc/demo/springboot/src/main/java/com/dahuatech/springboot/bean/Human.java
package com.dahuatech.springboot.bean;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.context.annotation.PropertySource;
import org.springframework.stereotype.Component;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Component
@ConfigurationProperties(prefix = "human")
@PropertySource("classpath:human.properties")// 如不指定则默认加载application.properties
public class Human {
    private String name;
    private int age;
}

File: scala/src/main/scala/com/dahuatech/scala/demo/Human.scala
package com.dahuatech.scala.demo

/**
 * 样例类创建模板，该模板创建的类不能通过反射创建，因为该类没有空参构造器
 * 参数不用val或者var修饰
 * 样例类是为模式匹配优化的类，通用性较差
 *
 * @param name
 * @param age
 */
case class Human(name: String, age: Int)
